core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/HttpSignRequest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.spi;
+
+import java.util.Optional;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.http.auth.spi.internal.DefaultHttpSignRequest;
+import software.amazon.awssdk.utils.builder.SdkBuilder;
+
+/**
+ * Represents a request to be signed by {@link HttpSigner}.
+ *
+ * @param <PayloadT> The type of payload of this request.
+ */
+@SdkProtectedApi
+@Immutable
+@ThreadSafe
+public interface HttpSignRequest<PayloadT> {
```

Comment:
Not a blocker, should it be `SignableHttpRequest` to match with`SignedHttpRequest`?

Reply:
This is as per design doc. cc'ing @millems. I think this is following the similar pattern for ResolveIdentityRequest used as parameter in `resolveIdentity()`.